### ISSUE TITLE
Users can use the same project link on all challenges and claim the cert, in regards to responsive design cert

### ISSUE THREAD

**Issue by JonDevOps** [freeCodeCamp/freeCodeCamp#35015]
**Describe your problem and how to reproduce it:**

Users can use the same project on all challenges and claim the cert, in regards to responsive design cert.
**Add a Link to the page with the problem:**
All responsive design projects.

**Tell us about your browser and operating system:**
* Browser Name: 
* Browser Version: 
* Operating System: 

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**

**Comment by raisedadead** [comment]
fCC uses an honor based academic honesty policy. The user profile has to be public, for certs to be visible.

This can hence be verified and reported back for dishonest work. We take such reports very seriously and  ban the profiles, and the certs.

Now coming to the specific use case of same URLs being used to abuse the system should be fixed. Thanks for the report.

**Comment by JonDevOps** [comment]
I just reported this because i saw my buddy do it..

**Comment by raisedadead** [comment]
Thanks @JonDevOps please ask them to remedy it to prevent getting flagged. Also, you can report such cases to `team@freecodecamp.org`

**Comment by Ronkiro** [comment]
Dupe of #34763 

It's a problem of all challenges there, not only the Responsive Design ones.

I also posted a solution there, that may help to not really solve, but to fix and add steps to help one's learning.

**Comment by raisedadead** [comment]
Quoting @Ronkiro 's proposal from another thread here:

> As a way to solve this, i think that creating basic questions and/or challenges that can acknowledge that you did it would be nice. It's just a step more to get the exercise completed, and can also help making one learn/consolidating what he learnt.

**Comment by raisedadead** [comment]
Requesting feedback from @freeCodeCamp/moderators

**Comment by lynxlynxlynx** [comment]
How about checking the URLs are unique when claiming the cert?

**Comment by fabiotp91** [comment]
I think the fair solution would be to make moderators check projects, but that would be insanely time consuming and people could still cheat by copying projects from other people. 

Checking the url's vs others that have been used before would be a good way to keep basic cheating away, of course there's always a way to cheat, but I think we should always keep FCC as open as possible without making the experience to the user too restrictive with a ton of rules.

**Comment by co16353sidak** [comment]
Exactly, Efforts shouldn't be on trying to prevent cheating, but more on the emphasis on how it is detrimental to the person doing it.
The effort should be to make it easier to learn and to aid someone looking to actually learn something as much as possible instead of looking to prevent cheaters.

**Comment by tbushman** [comment]
We are nearing the means to test the submitted projects thanks to the recent update from @ValeraS at freeCodeCamp/testable-projects-fcc . @raisedadead The gulpfile has the capability of pre-compiling jsx, sass, etc. We could bring that over to the main repo to build student html / css/ js and then link to the bundle.js for tests.

**Comment by raisedadead** [comment]
Hi, @tbushman 

I think that may not be of help. The testable projects tests are for testing codepen projects, that we recommend. But we do not mandate (by design) that codepen projects be necessary for the certification itself.

I think all we need here is an interactive way of confirming for genuine submissions like @Ronkiro proposed. And also URLs are unique for projects, which is a deterrent to spam as well.

**Comment by Bouncey** [comment]
There could be more social accountability where we pull in a screenshot of the url for GUI projects, not sure how we would handle api projects with this though.

**Comment by raisedadead** [comment]
> There could be more social accountability where we pull in a screenshot of the url for GUI projects

Yep. That'd be a great way to catch abuse.

**Comment by cmccormack** [comment]
The cert itself is the concern right?  Can we include a link or image (similar to the codepen preview) of the final project in the cert so duplicated URLs will be completely obvious?  Just an idea.

**Comment by JonDevOps** [comment]
@raisedadead I agree and I might add to your point that codepen is not mandatory by saying, I had trouble for awhile understanding codepen for months. Since there was no head or body tags i ended up being extremely confused on how to use it, and even to ths day still opt out and instead favor github and gh-pages because it's more comfortable, for me, to write code with every line of code being there. Though i do know how to use codepen now, and i  know plenty of beginners who actually ONLY know how to use codepen,I agree allowng people to submit projects not limited from just codepen is beneficial.

**Comment by QuincyLarson** [comment]
I agree with most of the people in this thread that there's not much we can do to *prevent* people from cheating.

Instead, we should focus on detecting cheating. We need a crawler of sorts that goes through people's submitted projects and automatically flags ones it thinks need human attention.

This can be open source and we can openly discuss ways to detect cheating. Since we would be continuously refining our cheating detection, cheating would be a losing battle (cheating results in permanent "flagged as cheater" status - even if we detect it months or years after the fact).

Would any of you be interested in helping with such a cheating detection project?

**Comment by JonDevOps** [comment]
I would love to. I'm currently going through basic JS so I know I wouldn't be the best of help but I want to keep track of this and this can be my first dive intomore advanced topics than html and css. Also I have a learning group and a cohort group Mr. Larson with hundreds of people ...and we start tomorrow :) can i get them on this somehow? We have data scientist js devs, etc. Can I zoom with you and show you that stats and get your opinion on how we can mobilize their efforts? I would like to learn as we go but some of the others would probably be up to par. Let me know I want to learn, help, and help others learn. BTW all my members are freeCodeCamp students and lovers to the core. Alot came from the fCC reddit, twitter, or my other socials. 

TL;DR I'm in I only know html and css but i planned on starting javascript with my learning group tomorrow anyway. My build to learn team members also need something to do so this would be a nice project for whoever has the skills already. Everyone wants to learn and loves FCC :)

**Comment by QuincyLarson** [comment]
I'm closing this issue in favor of a dedicated discussion around these tools: https://github.com/freeCodeCamp/freeCodeCamp/issues/35054
